### DISCUSSION TITLE
StateTimeLine: Show missing data or do not always extend boxes to end of time window

### DISCUSSION THREAD

**Discussion by baur** [grafana/grafana#38296] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

I see the status of future time

![image](https://user-images.githubusercontent.com/2898361/121625939-46671d00-ca96-11eb-9293-2becce14380c.png)

Although the data has not yet appeared

![image](https://user-images.githubusercontent.com/2898361/121626229-ddcc7000-ca96-11eb-90dd-289299dcd1bb.png)

**Why is this needed**:

**Would be nice if StateTimeLine redner only existing data**

PS:
By the way, the clock panel background color is not working

**Comment by torkelo** [comment]
The time points in the state timeline are just the start of the state, it assumes the state persists until the next state. You can switch to status history if you have periodic data and want state boxes to only show centered on the values.

**Comment by torkelo** [comment]
A possible solution would be to have a gap option with an allowed time window, if no value in the specified gap window an unknown state will be added to graph?

**Comment by pepjo** [comment]
We would also like this feature!

**Comment by Legion2** [comment]
We use the State timeline to visualize device heartbeats. However if a device did not send a heartbeat for a long time its state is unknown and should also be visualized in State timeline as unknown instead of showing the previous state. A configurable gap or timeout duration would be optimal.

To visualize a gap there would be two options:
 1. if the gap is greater than the maximum the states are not connected and nothing is drawn between the points.
 2. each data point represents a state area that extends from its timestamp until the maximum duration.

For example option 1 does not draw any area between points if they are too far apart. Option 2 will draw an area in the size of the maximum duration if the data points are too far apart or if it is the latest data point in the series.

Option 2 can also be extended to allow the configuration of the alignment of data points similar to the Bar alignment of time series.

**Comment by leeoniya** [comment]
will likely be part of https://github.com/grafana/grafana/pull/36622

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
